File: frontend/src/services/userService.js
import api from './api';

const userService = {
  updateProfile: async (userId, data) => {
    const response = await api.patch(`/users/${userId}`, data);
    return response.data;
  },

  getCurrentUser: async () => {
    const response = await api.get('/users/me');
    return response.data;
  },

  getUserById: async (userId) => {
    const response = await api.get(`/users/${userId}`);
    return response.data;
  },
};

export default userService;

File: src/main/java/chatterbox/services/UserService.java
package chatterbox.services;

import chatterbox.dtos.request.UpdateUserProfileDto;
import chatterbox.dtos.response.UserDto;
import java.util.List;

public interface UserService {
  List<UserDto> getAllUsers();

  UserDto getCurrentUser(Long currentUserId);

  UserDto getUserById(Long userId);

  void updateUserProfile(Long userId, UpdateUserProfileDto dto);
}
